gradle-plugin/src/main/java/com/squareup/anvil/plugin/AnvilSubplugin.kt

```diff
@@ -50,10 +53,17 @@ class AnvilSubplugin : KotlinGradleSubplugin<AbstractCompile> {
       // Kotlin Gradle plugin adds all SubpluginOptions as input to the Gradle task. This is bad,
       // because now the hash of inputs is different on every machine. We override this input with
       // a relative path in order to preserve some safety and avoid the cache misses.
-      kotlinCompile.inputs.property(
-          "${getCompilerPluginId()}.${srcGenDirOption.key}",
-          srcGenDir.relativeTo(project.buildDir).path
-      )
+      val key = "${getCompilerPluginId()}.${srcGenDirOption.key}"
+      val relativePath = srcGenDir.relativeTo(project.buildDir).path
+
+      project.tasks.withType(KotlinCompile::class.java) { task ->
+        task.inputs.property(key, relativePath)
+      }
+      project.tasks.withType(KaptTask::class.java) { task ->
+        // We don't apply the compiler plugin for KaptTasks. But for some reason the Kotlin Gradle
+        // plugin copies the inputs with the "kotlinCompile" prefix.
+        task.inputs.property("kotlinCompile.$key", relativePath)
+      }
```

Comment:
This used to be way more dynamic, but I was seeing strange build errors in CI only. No idea what happened.
```kotlin
      val key = "${getCompilerPluginId()}.${srcGenDirOption.key}"

      project.tasks
          .filter { it is KotlinCompile || it is KaptTask }
          .forEach { task ->
            task.inputs.properties
                .filter { it.key.contains(key) && it.value == srcGenDirPath }
                .forEach {
                  task.inputs.property(it.key, srcGenDir.relativeTo(project.buildDir).path)
                }
          }
```
This new solution is goo enough.